### PR TITLE
Feature/nf 1146 group UI improvement

### PR DESCRIPTION
**Pull request by jitendra-simform**
Create chat header view updated ,
chat user list code updated 
Search and Group image option removed from group chat menu

### CODE REVIEW

File: Sources/StreamChatUI/Controller/ChatGroupDetailsVC.swift [hunk 1]
```diff
@@ -103,12 +128,11 @@ public class ChatGroupDetailsVC: ChatBaseVC {
     @IBAction func backBtnTapped(_ sender: UIButton) {
         popWithAnimation()
     }
-    @IBAction func doneTapped(_ sender: UIButton) { }
```
Comment (parth-simformsolutions) [inline Sources/StreamChatUI/Controller/ChatGroupDetailsVC.swift:106]: @jitendra-simform  Please add one line after function or method

File: Sources/StreamChatUI/Controller/ChatGroupDetailsVC.swift [hunk 2]
```diff
@@ -140,102 +164,93 @@ public class ChatGroupDetailsVC: ChatBaseVC {
...
-                break
-            }
-            self.view.layoutIfNeeded()
+    @objc func showMoreButtonAction(_ sender: UIButton) {
+        self.buttonShowMore.isHidden = true
+        let visibleRow = IndexPath.init(row: self.usersCount, section: 0)
+        self.usersCount = self.selectedUsers.count
+        self.tableView.reloadData()
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
+            self.tableView.scrollToRow(at: visibleRow, at: .bottom, animated: true)
```
Comment (parth-simformsolutions) [inline Sources/StreamChatUI/Controller/ChatGroupDetailsVC.swift:180]: @jitendra-simform  Please add weak self here.

File: Podfile [hunk 3]
```diff
@@ -86,6 +85,7 @@ target 'StreamChatUI' do
   # Comment the next line if you don't want to use dynamic frameworks
   use_frameworks!
   pod 'swiftScan', :git => 'https://github.com/MxABC/swiftScan.git'
+  pod 'SkeletonView'
```
Comment (ajay-simform) [inline Podfile:88]: @jitendra-simform  Please add pod version.

File: Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift [hunk 4]
```diff
@@ -851,7 +854,9 @@ open class ChatChannelVC:
                 let isAdmin = channelController?.channel?.createdBy?.id == ChatClient.shared.currentUserId
                 if isAdmin {
                     var actions: [UIAction] = []
-                    actions.append(contentsOf: [groupImage, search,invite,groupQR])
+                    // To do:- will add in future release
+                    //actions.append(contentsOf: [groupImage, search,invite,groupQR])
+                    actions.append(contentsOf: [invite,groupQR])
```
Comment (ajay-simform) [inline Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift:859]: @jitendra-simform add space between , and groupQR

File: Sources/StreamChatUI/CustomView/UserListDataSource.swift [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+//  UserListDataSource.swift
+//  StreamChat
+//
+//  Created by Jitendra Sharma on 25/02/22.
+//  Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public class UserListDataSource: NSObject {
```
Comment (ajay-simform) [inline Sources/StreamChatUI/CustomView/UserListDataSource.swift:11]: @jitendra-simform  please remove UserListDataSource.

File: Sources/StreamChatUI/Model/UserListViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+// MARK: - SORT METHODS
+extension UserListViewModel {
+    public func sortAtoZ(filteredUsers: [ChatUser]) -> ChatUserListData {
+        let alphabetUsers = filteredUsers.filter { ($0.name?.isFirstCharacterAlp ?? false) && $0.name?.isBlank == false }.sorted{ $0.name!.localizedCaseInsensitiveCompare($1.name!) == ComparisonResult.orderedAscending}
+        let otherUsers = filteredUsers.filter { ($0.name?.isFirstCharacterAlp ?? false) == false }.sorted{ $0.id.localizedCaseInsensitiveCompare($1.id) == ComparisonResult.orderedAscending}
+        var data = ChatUserListData.init(letter: "", sectionType: .noHeader)
+        data.users = alphabetUsers
+        data.users.append(contentsOf: otherUsers)
+        return data
+    }
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Model/UserListViewModel.swift:61]: Add one line space

File: Sources/StreamChatUI/Model/UserListViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+        let alphabetUsers = onlineUser.filter { ($0.name?.isFirstCharacterAlp ?? false) }
+        let nonAlphabetUsers = onlineUser.filter { ($0.name?.isFirstCharacterAlp ?? false) == false}
+        let otherUsers = filteredUsers.filter({ $0.isOnline == false && $0.name?.isBlank == false}).sorted(by: { ($0.lastActiveAt ?? $0.userCreatedAt) > ($1.lastActiveAt ?? $1.userCreatedAt )})
+        var data = ChatUserListData.init(letter: "", sectionType: .noHeader)
+        data.users = alphabetUsers
+        data.users.append(contentsOf: nonAlphabetUsers)
+        data.users.append(contentsOf: otherUsers)
+        return data
+    }
+    public func shortByName(filteredUsers: [ChatUser]) -> [ChatUserListData] {
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Model/UserListViewModel.swift:73]: @jitendra-simform  add one line spacing here

File: Sources/StreamChatUI/Model/UserListViewModel.swift [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+                .autocomplete(.name, text: strName),
+                .exists(.lastActiveAt),
+                .notEqual(.id, to: ChatClient.shared.currentUserId ?? ""),
+            ])
+            searchListController.search(query: newQuery) { [weak self] error in
+                guard let weakSelf = self else { return }
+                if let error = error {
+                    weakSelf.dataLoadingState = .searchingError
+                } else {
+                    let filterData = weakSelf.searchListController.users.filter { $0.name?.isEmpty == false }.filter { $0.id.isEmpty == false }.filter { $0.id != ChatClient.shared.currentUserId ?? "" }
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Model/UserListViewModel.swift:123]: @jitendra-simform  merge these filters in single one

File: Sources/StreamChatUI/Model/UserListViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+        if fetchMoreData {
+            self.dataLoadingState = .loadMoreData
+            var userQuery = UserListQuery(filter: .and([
+                .notEqual(.id, to: ChatClient.shared.currentUserId ?? ""),
+            ]), sort: [], pageSize: 99)
+            self.userListController = ChatClient.shared.userListController(query: userQuery)
+            self.userListController.synchronize { [weak self] error in
+                guard let weakSelf = self else { return }
+                if error == nil {
+                    let filterData = weakSelf.userListController.users.filter { $0.name?.isEmpty == false }.filter { $0.id.isEmpty == false }.filter { $0.id != ChatClient.shared.currentUserId ?? "" }
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Model/UserListViewModel.swift:150]: @jitendra-simform merge these filter in single one

File: Sources/StreamChatUI/Model/UserListViewModel.swift [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+                weakSelf.dataLoadingState = .error
+            }
+        } else {
+            let date = Calendar.current.date(byAdding: .day, value: -1, to: Date())!
+            let userQuery = UserListQuery.init(filter: .greaterOrEqual(.lastActiveAt, than: date), sort: [], pageSize: 99)
+            self.userListController.query = userQuery
+            self.userListController.synchronize { [weak self] error in
+                guard let weakSelf = self else { return }
+                if error == nil {
+                    let filterData = weakSelf.userListController.users.filter { $0.name?.isEmpty == false }.filter { $0.id.isEmpty == false }.filter { $0.id != ChatClient.shared.currentUserId ?? "" }
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Model/UserListViewModel.swift:164]: @jitendra-simform merge these filter in single one

File: Sources/StreamChatUI/Model/UserListViewModel.swift [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+                    return
+                }
+                weakSelf.dataLoadingState = .error
+            }
+        }
+    }
+    
+    open func sortUserList() {
+        if let strName = searchText, strName.isBlank == false {
+            let filterData = self.searchListController.users.filter { $0.name?.isEmpty == false }.filter { $0.id.isEmpty == false }.filter { $0.id != ChatClient.shared.currentUserId ?? "" }
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Model/UserListViewModel.swift:176]: @jitendra-simform  merge these filter in single one

File: Sources/StreamChatUI/Model/UserListViewModel.swift [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+            self.bCallbackDataUserList?(filterData)
+        } else  {
+            let filterData = self.userListController.users.filter { $0.name?.isEmpty == false }.filter { $0.id.isEmpty == false }.filter { $0.id != ChatClient.shared.currentUserId ?? "" }
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Model/UserListViewModel.swift:179]: @jitendra-simform  merge these filter in single one

File: Sources/StreamChatUI/Model/UserListViewModel.swift [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+}
+// MARK: - Chat user controller delegate
+extension UserListViewModel: ChatUserListControllerDelegate {
+    public func controller(_ controller: ChatUserListController, didChangeUsers changes: [ListChange<ChatUser>]) {
+        // To Do
+    }
+}
+public struct DTFormatter {
+    public static var formatter: DateFormatter = {
+        let formatter = DateFormatter()
```
Comment (ajay-simform) [inline Sources/StreamChatUI/Model/UserListViewModel.swift:192]: @jitendra-simform can't we move dateFormatter to DateFormatter+Extension

File: Sources/StreamChatUI/xibs/cells/TableViewCellChatUser.swift [hunk 14]
```diff
@@ -23,13 +22,26 @@ public class TableViewCellChatUser: UITableViewCell {
     @IBOutlet public var lblRole: UILabel!
     // MARK: - Variables
     private var user: ChatUser?
-    //
-    // MARK: - Functions
-    public func config(user: ChatUser, selectedImage: UIImage?, avatarBG: UIColor) {
+    private let shimmerBackgroundColor = Appearance.default.colorPalette.placeHolderBalanceBG
+    private lazy var shimmerGradient = SkeletonGradient(colors: [shimmerBackgroundColor.withAlphaComponent(0.3),shimmerBackgroundColor.withAlphaComponent(0.5), shimmerBackgroundColor.withAlphaComponent(0.3)])
```
Comment (ajay-simform) [inline Sources/StreamChatUI/xibs/cells/TableViewCellChatUser.swift:26]: @jitendra-simform separate out skeletonGradient parameter to new line

File: Sources/StreamChatUI/xibs/cells/TableViewCellGroupDetailsAttachmentsList.swift [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+            case 0:
+                self.indicatorViewLeadingContraint.constant = self.buttonMedia.frame.origin.x
+            case 1:
+                self.indicatorViewLeadingContraint.constant = self.buttonFiles.frame.origin.x
+            case 2:
+                self.indicatorViewLeadingContraint.constant = self.buttonLinks.frame.origin.x
+            default:
+                break
+            }
+
```
Comment (ajay-simform) [inline Sources/StreamChatUI/xibs/cells/TableViewCellGroupDetailsAttachmentsList.swift:87]: @jitendra-simform  remove line spacing

**Review by ajay-simform — COMMENTED** [review]
General comment:
- @jitendra-simform  Please do check at your side before PR creation. Many silly mistakes can be found easily.
- use self. only when required.

**Pull request merged**